### PR TITLE
added info about testing and reformatted how to follow releases

### PR DESCRIPTION
**Pull request by thepomeranian**
### <samp>🤖 Generated by Copilot at d259665</samp>

This pull request enhances the release documentation for Aptos by updating the `RELEASE.md` file and the `developer-docs-site/docs/releases/index.md` file. It clarifies the branching and tagging strategy, adds links to useful resources, and provides more options and details on how to subscribe to release announcements via GitHub, Discord, and Twitter.

### CODE REVIEW

**Review by gregnazario — APPROVED** [review]

**Review by thepomeranian — COMMENTED** [review]
made some grammar and spelling edits :( english is hard sometimes yo

**Review by saharct — APPROVED** [review]
@thepomeranian, LGTM with some changes!

File: RELEASE.md [hunk 1]
```diff
@@ -17,32 +17,34 @@
...
-the `main` branch. Release branches are created on every 1-2 months cadence.
+the `main` branch every 1-2 months.
 
 ### aptos-node-v*X.Y.Z* release tag
-The aptos node release tags are created for validator/fullnode deployment of the given release
-branch. The minor number *Z* will increment when a new hot-fix release is required on this 
-release branch. Aptos team will publish the matching tag docker images on 
-[Aptos Docker Hub](https://hub.docker.com/r/aptoslabs/validator/tags) when available.
+The aptos node release tags are created for validator/fullnode deployment of the given release branch. The minor number *Z* will increment when a new hot-fix release is required on the release branch. Aptos team will publish the matching tag docker images on 
+[Aptos Docker Hub](https://hub.docker.com/r/aptoslabs/validator/tags) when it's available.
```
Comment (davidiw) [inline RELEASE.md:31]:
```suggestion
The aptos node release tags are created for validator/fullnode deployment of the given release branch. The minor number *Z* will increment when a new hot-fix release is required on the release branch. Aptos team will publish the matching tag docker images on [Aptos Docker Hub](https://hub.docker.com/r/aptoslabs/validator/tags) when it's available.
```

**Review by davidiw — APPROVED** [review]
amazing

**Pull request merged**